Content.Server/Weapons/Ranged/Systems/GunSystem.cs

```diff
@@ -396,23 +396,24 @@ private void FireEffects(EntityCoordinates fromCoordinates, float distance, Angl
         // Forgive me for the shitcode I am about to do
         // Effects tempt me not
         var sprites = new List<(NetCoordinates coordinates, Angle angle, SpriteSpecifier sprite, float scale)>();
-        var gridUid = fromCoordinates.GetGridUid(EntityManager);
+        var fromXform = Transform(fromCoordinates.EntityId);
         var angle = mapDirection;
 
         // We'll get the effects relative to the grid / map of the firer
         // Look you could probably optimise this a bit with redundant transforms at this point.
-        var xformQuery = GetEntityQuery<TransformComponent>();
 
-        if (xformQuery.TryGetComponent(gridUid, out var gridXform))
+        var gridUid = fromXform.GridUid;
+        if (gridUid != fromCoordinates.EntityId && TryComp(gridUid, out TransformComponent? gridXform))
         {
```

Comment:
Didn't all entities have a TransformComponent? So no need to do a TryComp
```suggestion
        if (gridUid != fromCoordinates.EntityId)
        {
            var gridXform = Transform(gridUid);
```

Reply:
gridUid might be null here, so the `TryComp` also acts as a ` != null` check while simultaneously getting the component and I don't think it makes it any harder to read